### DISCUSSION TITLE
How to setup build and run like Geany?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#155] [General]
For example, with C++. Thank you.

**Comment by SpartanJ** [comment]
The release video displaying the feature might help you:

https://github.com/SpartanJ/ecode/assets/650416/5ec8fe75-a278-4fe7-be84-e786952848d3

**Reply by ghost** [reply]
Is it possible to use this feature on legacy Windows? It's depending on the terminal feature that is not available on legacy Windows.

**Reply by SpartanJ** [reply]
Yes, you shouldn't have any problem, it doesn't use the terminal.

**Reply by ghost** [reply]
This feature is only about build, isn't it? How to run the compiled program? Geany has a button to run the compiled program.

**Reply by ghost** [reply]
Btw, in Geany, `%f` is the name of the current file you are working on and `%e` is the name of that file but without extension. Is there something equivalent in ecode?

**Reply by SpartanJ** [reply]
> This feature is only about build, isn't it? How to run the compiled program? Geany has a button to run the compiled program.

Yes, only build. The run program option will be added when the debugger plugin is implemented. I understand that in your case this would be useful because you don't have access to the console, so it could be handy for you.

> Btw, in Geany, %f is the name of the current file you are working on and %e is the name of that file but without extension. Is there something equivalent in ecode?

Ohh, not it's not because it's designed to call directly to a build script. But it makes total sense to give you access to that. I'll add it! If you expand the advanced options you'll see the default custom variables available (${project_root}, ${build_type}, ${os}, ${nproc}). And you can add any custom variable, but this one is contextual so I'll have to add it myself.

**Reply by mhe98xmuh** [reply]
> The release video displaying the feature might help you:
> 
>  252847447-09bb03cb-91f0-4350-8bec-a48d6f06bdd2.mp4

Is it the same when it is done?Mingw On windows?

**Reply by SpartanJ** [reply]
Same logic. It will depend on your mingw distribution which commands you'll need to run.

**Reply by mhe98xmuh** [reply]
> Same logic. It will depend on your mingw distribution which commands you'll need to run.

Actually, I'm a beginner and I liked it Ecode I still use Windows Can you give me the required steps?

**Reply by SpartanJ** [reply]
I don't know what you want to build. What do you want to build?
That video is just a demonstration of the feature, I can't tell you what to do if I don't know what you want to do.
